lib/computed_model.rb

```diff
@@ -145,27 +148,25 @@ def delegate_dependency(*methods, to:, allow_nil: nil, prefix: nil, include_subd
     # or set `computed_model_error` otherwise.
     #
     # @param meth_name [Symbol] the name of the loaded attribute.
+    # @param key [Proc] The proc to collect keys.
     # @return [void]
-    # @yield [objects, **options]
-    # @yieldparam objects [Array] The objects to preload the attribute into.
+    # @yield [keys, subdeps, **options]
+    # @yieldparam objects [Array] The ids of the loaded attributes.
+    # @yieldparam subdeps [Hash] sub-dependencies
     # @yieldparam options [Hash] A verbatim copy of what is passed to {#bulk_load_and_compute}.
-    # @yieldreturn [void]
+    # @yieldreturn [Array]
```

Comment:
So the order of the array matters, right? I'm a little concerned about the fragility, like directly returning `Something.where(user_id: user_ids).to_a`.

How about returning instead a hash of keys and loaded values?

Reply:
fixed in 9cb446d